### PR TITLE
711 enable disable mechanism

### PR DESCRIPTION
**Pull request by ShiriMoran**

### CODE REVIEW

File: cmd/analyzer/subcmds/lint.go [hunk 1]
```diff
@@ -7,32 +7,118 @@ SPDX-License-Identifier: Apache-2.0
...
+	return strings.Join(legalNamesSlice, ",")
+}
+
+func bothDisableAndEnable(enableLintersList, disableLintersList []string) error {
+	for _, enable := range enableLintersList {
+		for _, disable := range disableLintersList {
+			if enable == disable {
+				return fmt.Errorf("lint %s specified both as enable and disable. only one is possible", enable)
+			}
+		}
```
Comment (zivnevo) [inline cmd/analyzer/subcmds/lint.go:121]:
```suggestion
		if slices.Contains(disableLintersList, enable) {
			return fmt.Errorf("lint %s specified both as enable and disable. only one is possible", enable)
		}
```

File: cmd/analyzer/subcmds/lint.go [hunk 2]
```diff
@@ -7,32 +7,118 @@ SPDX-License-Identifier: Apache-2.0
...
+			return validateLintFlags(cmd, args)
+		},
 		RunE: func(cmd *cobra.Command, _ []string) error {
 			return lintVPCConfigs(cmd, args)
 		},
 	}
+	validLintersNames := getListLintersName(linter.GetLintersNames())
+	usageStr := "specific linters " + enableDisableUsage + " linters: " + validLintersNames
+	cmd.Flags().StringSliceVar(&args.enableLinters, enable, []string{}, enable+space+usageStr)
+	cmd.Flags().StringSliceVar(&args.disableLinters, disable, []string{}, disable+space+usageStr)
```
Comment (zivnevo) [inline cmd/analyzer/subcmds/lint.go:41]:
```suggestion
	usageStr := " specific linters, specified as linter names separated by comma.\nlinters: " + validLintersNames
	cmd.Flags().StringSliceVar(&args.enableLinters, enable, []string{}, enable+usageStr)
	cmd.Flags().StringSliceVar(&args.disableLinters, disable, []string{}, disable+usageStr)
```

File: cmd/analyzer/subcmds/lint.go [hunk 3]
```diff
@@ -7,32 +7,118 @@ SPDX-License-Identifier: Apache-2.0
...
+	if errEnable != nil {
+		return errEnable
+	}
+	disableList, errDisable := validateLintEnableOrDisable(cmd, disable)
+	if errDisable != nil {
+		return errDisable
+	}
+	if errBothEnableDisable := bothDisableAndEnable(enableList, disableList); errBothEnableDisable != nil {
+		return errBothEnableDisable
+	}
```
Comment (zivnevo) [inline cmd/analyzer/subcmds/lint.go:75]:
```suggestion
	errEnable := validLintersName(args.enableLinters, enable)
	if errEnable != nil {
		return errEnable
	}
	errDisable := validLintersName(args.disableLinters, disable)
	if errDisable != nil {
		return errDisable
	}
	if errBothEnableDisable := bothDisableAndEnable(args); errBothEnableDisable != nil {
		return errBothEnableDisable
	}
```

File: cmd/analyzer/subcmds/lint.go [hunk 4]
```diff
@@ -7,32 +7,118 @@ SPDX-License-Identifier: Apache-2.0
...
-	multiConfigs, err1 := buildConfigs(inArgs)
+	multiConfigs, err1 := buildConfigs(args)
 	if err1 != nil {
 		return err1
 	}
-	_, _, err2 := linter.LinterExecute(multiConfigs.Configs())
+	// potential errors already handled
+	enableList, _ := cmd.Flags().GetStringSlice(enable)
+	disableList, _ := cmd.Flags().GetStringSlice(disable)
+	_, _, err2 := linter.LinterExecute(multiConfigs.Configs(), enableList, disableList)
```
Comment (zivnevo) [inline cmd/analyzer/subcmds/lint.go:56]:
```suggestion
	_, _, err2 := linter.LinterExecute(multiConfigs.Configs(), args.enableLinters, args.disableLinters)
```

File: pkg/linter/linterExecute.go [hunk 5]
```diff
@@ -17,9 +17,25 @@ import (
...
+		ruleNonRelevantCIDRSGName: true}
+}
+
 // LinterExecute executes linters one by one
-// todo: mechanism for disabling/enabling lint checks
-func LinterExecute(configs map[string]*vpcmodel.VPCConfig) (issueFound bool, resString string, err error) {
+func LinterExecute(configs map[string]*vpcmodel.VPCConfig,
+	enableList, disableList []string) (issueFound bool, resString string, err error) {
+	enableLinters := // enabling and disabling linters defaults
+		map[string]bool{blockedTCPResponse: true, splitRuleSubnetNACLName: true, splitRuleSubnetSGName: true,
```
Comment (zivnevo) [inline pkg/linter/linterExecute.go:30]: Didn't we say `splitRuleSubnetSGName` should be off by default?

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: cmd/analyzer/subcmds/lint.go (hunk 3 above)
Comment (ShiriMoran) [reply cmd/analyzer/subcmds/lint.go]: done

File: pkg/linter/linterExecute.go (hunk 5 above)
Comment (ShiriMoran) [reply pkg/linter/linterExecute.go:30]: Yes, we did. Changed default.

File: cmd/analyzer/subcmds/lint.go [hunk 6]
```diff
@@ -7,32 +7,103 @@ SPDX-License-Identifier: Apache-2.0
...
 	"github.com/spf13/cobra"
 
 	"github.com/np-guard/vpc-network-config-analyzer/pkg/linter"
 )
 
+const (
+	enable             = "enable"
+	disable            = "disable"
+	enableDisableUsage = "specified as linters names separated by comma"
+	space              = " "
```
Comment (zivnevo) [inline cmd/analyzer/subcmds/lint.go:23]:
```suggestion
```
unused

**Review by zivnevo — CHANGES_REQUESTED** [review]
Just this tiny thing

**Review by zivnevo — APPROVED** [review]

**Pull request merged**